client/src/app/common/generic-clear-memory.ts

```diff
@@ -0,0 +1,83 @@
+import { GenericClearMemoryModalComponent } from './generic-clear-memory-modal/generic-clear-memory-modal.component';
+import { MatDialog } from '@angular/material';
+import { Utilities } from './utlilities';
+import { TsScreenerDataService } from '../services/ts-screener-data.service';
+import { ThsDataService } from '../services/ths-data.service';
+import { TfiDataService } from '../services/tfi-data.service';
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export default class GenericClearMemory {
+    private sessionKeys: string[] = [
+        'patient-id',
+        'tests-data',
+        'tfi-dataRecord',
+        'tfi-currentState',
+        'ths-dataRecord',
+        'ths-history',
+        'ths-currentState',
+        'ts-dataRecord',
+        'ts-history',
+        'ts-currentState',
+        'nextComponent',
+        'last-name',
+        'email',
+        'first-name',
+        'appt'
+    ];
+
+    constructor(
+        private dialog: MatDialog,
+        private tsDataService: TsScreenerDataService,
+        private thsDataService: ThsDataService,
+        private tfiDataService: TfiDataService) { }
+
+        /**
+         * Call this anytime you want to clear all patient/appointment data stored in client services or sessionMemory.
+         * If there is no data to clear, this will call onSuccess().
+         * @param warn Whether to display the modal asking the user to confirm clear memory
+         * @param onSuccess Callback if memory is cleared, called if warn is false, or if user selects 'continue'
+         * @param onCancel Callback if user selected 'cancel' so memory was not cleared.
+         */
+    public clearMemory(warn: boolean, onSuccess: () => void, onCancel?: () => void): void {
```

Comment:
Optional: It might make more sense to have warn always be true and rename this method to `askToClearData`, and allow callers that don't want a warning to simply call `clearData`.